jupyter-js-widgets/src/widget_int.js

```diff
@@ -200,23 +203,43 @@ var IntSliderView = widget.DOMWidgetView.extend({
             var readout = this.model.get('readout');
             if (readout) {
                 this.readout.style.display = '';
+                window.setTimeout(function() {
+                    if (that.readout_overflow()) {
+                        that.readout.classList.add('overflow');
+                    } else {
+                        that.readout.classList.remove('overflow');
+                    }
+                });
             } else {
                 this.readout.style.display = 'none';
             }
+
         }
         return IntSliderView.__super__.update.apply(this);
     },
 
     /**
+     * Readout purpling
+     * Change the border color of the readout if the text overflows
+     *
+     */
+    readout_overflow: function() {
+        return this.readout.scrollWidth > this.readout.clientWidth;
+    },
+
+    /**
      * Write value to a string
      * @param  {number|number[]} value
      * @return {string}
      */
     valueToString: function(value) {
+        var format = d3format(this.model.get('readout_format'));
```

Comment:
I think the format function should be updated and cached as an attribute instead of recreated each time it's used.